pkg/apps/apis/apps/validation/validation.go

```diff
@@ -39,7 +39,12 @@ func ValidateDeploymentConfigSpec(spec appsapi.DeploymentConfigSpec) field.Error
 		podSpec = &spec.Template.Spec
 	}
 
-	allErrs = append(allErrs, validateDeploymentStrategy(&spec.Strategy, podSpec, specPath.Child("strategy"))...)
+	// gate calls to validation functions that depend on a podSpec
+	// if we have don't have an actual podSpec here.
+	if podSpec != nil {
+		allErrs = append(allErrs, validateDeploymentStrategy(&spec.Strategy, podSpec, specPath.Child("strategy"))...)
+	}
```

Comment:
In addition, not having a podSpec should be an error on it's own but I don't see it checked anywhere else

Reply:
It is already validated in https://github.com/openshift/origin/blob/master/pkg/apps/apis/apps/validation/validation.go#L59-L66